File: Lista_Exercicio_Trabalho/Lista_2/exercicio_1/utils/validate.py
from view.view import *

def getNote(msg = '<none>'):
    while True:
        try:
            num = str(input(msg))
            
            if(not num.isdigit()):
                error('Valor digitado inválido. Tente novamente')
                continue

            else:
                num = int(num)
                if (num <= 0):
                    error('Digite um valor acima de 0')
                    continue


        except(ValueError, TypeError):
            error('Valor digitado inválido. Tente novamente')
            continue

        except(KeyboardInterrupt):
            error('Campo não pode ficar vazio')
            continue

        else:
            return int(num)
        

def getName(msg = '<none>'):
    while True:
        try:
            name = str(input(msg)).strip()

            if not name.isalpha():
                error('So é válido letras. Tente novamente')
                continue

            elif name == '':
                error('Campo não pode ficar vazio. Tente novamente')
                continue

        except:
            error('Error inesperado. Tente novamente')
            continue

        else:
            return name


def validateOut(opc = None):
    '''
    Retorna FALSE se vai continuar e TRUE se for parar
    '''
    while True:
        menuOpc()

        opc = str(input()).upper()

        if opc =='X':
            return True
        
        elif opc == '':
            return False

        else:
            error('Opção inválida. Tente novamente')
            
        

File: Lista_Exercicio_Trabalho/Lista_1/Exercicio_2-Lista-de-Compras/src/view/validate.ts
export const request = require("prompt-sync")();
import { clear } from "./visual";

export function getOption(list: Array<number>): number {
  for (let i = 0; i == 0; ) {
    try {
      console.log("-------------------------------------------");
      const option: number = Number(request("Digite sua opção: "));
      clear();
      if (list.includes(option)) {
        return option;
      } else {
        console.log("Error. Opção inválida. Tente novamente");
        continue;
      }
    } catch (e) {
      if (e instanceof TypeError) {
        clear();
        console.log("-------------------------------------");
        console.log("Error. Tipo inválido. Tente novamente");
        continue;
      }
    }
  }
  return -1;
}

export function getAccount(message: string = ""): number {
  for (let i = 0; i == 0; ) {
    const account: number = Number(request(message));
    clear();
    if (isNaN(account)) {
      console.log("------------------------------------");
      console.log("Error. Valor digitado inválido. Tente novamente.");
      continue;
    } else if (account == 0) {
      console.log("------------------------------------");
      console.log("Error. Campo não pode ficar vazio. Tente Novamente.");
    } else {
      return account;
    }
  }
  return -1;
}

export function alphabeticalOrder(
  array: Array<number | string>[]
): Array<number | string>[] {
  array.sort();
  return array;
}
